### PR TITLE
feat(ci): increase timeout for generating flamegraphs

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

after #9560 timeout increased for flamegraph generation, since more profiling data.

## Checklist For Contributors

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: ci/workflows/main-cron.yml [hunk 1]
```diff
@@ -307,7 +307,7 @@ steps:
     # - Nexmark event generation: ~3min for 100mil records.
     # - Generate Flamegraph: ~15min (see https://github.com/koute/not-perf/issues/30 on optimizing)
     # - Building RW artifacts: ~8min
-    timeout_in_minutes: 360
+    timeout_in_minutes: 480
```
Comment (kwannoel) [inline ci/workflows/main-cron.yml:310]: +2h

**Comment by kwannoel** [comment]
Actually... Let me change this to run weekly instead. I don't think we really need daily flamegraph. We can trigger it adhoc if needed.

File: ci/workflows/gen-flamegraph.yml [hunk 2]
```diff
@@ -15,17 +15,6 @@ cargo-cache: &cargo-cache
...
-    if: build.env("CPU_FLAMEGRAPH") == "true" || build.env("HEAP_FLAMEGRAPH") == "true"
-    plugins:
-      - chronotc/monorepo-diff#v2.3.0:
-          diff: "git diff --name-only origin/main"
-          watch:
-            - path: "ci/build-ci-image.sh"
-              config:
-                command: "ci/build-ci-image.sh"
-                label: "ci build images"
-  - wait
```
Comment (kwannoel) [inline ci/workflows/gen-flamegraph.yml:28]: I don't think this is needed

**Comment by kwannoel** [comment]
> Actually... Let me change this to run weekly instead. I don't think we really need daily flamegraph, kind of expensive. We can trigger it adhoc if needed.

Screenshot 2023-05-04 at 11 06 31 AM

It will run weekly on sun.

**Comment by kwannoel** [comment]
Ready for review.

**Comment by kwannoel** [comment]
We can adjust the cron setting in `buildkite`, so splitting it up is still beneficial. Can always revert back to daily if needed. Additionally, it won't block buildkite notification for `main-cron`, which happens much later as a result of waiting for generate flamegraph to complete.

**Review by lmatz — APPROVED** [review]

**Pull request merged**